poetry/console/commands/shell.py

```diff
@@ -0,0 +1,52 @@
+from os import environ
+from distutils.util import strtobool
+
+from .venv_command import VenvCommand
+
+
+class ShellCommand(VenvCommand):
+    """
+    Spawns a shell within the virtual environment.
+
+    shell [options]
+    """
+
+    help = """The <info>shell</> command spawns a shell, according to the
+<comment>$SHELL</> environment variable, within the virtual environment.
+If one doesn't exist yet, it will be created.
+"""
+
+    def handle(self):
+        # Check if it's already activated or doesn't exist and won't be created
+        if strtobool(environ.get("POETRY_ACTIVE", "0")) or not self.venv.is_venv():
+            current_venv = environ.get("VIRTUAL_ENV")
+
+            if current_venv:
+                self.line(
+                    "Virtual environment already activated: "
+                    "<info>{}</>".format(current_venv)
+                )
+
+            else:
+                self.error("Virtual environment wasn't found")
+
+            return
+
+        if self.venv._windows:
+            # Windows is probably more complicated than this
+            # and support to cmd alternatives should probably be added
+            shell = "cmd"
+            args = ["/k"]
+        else:
+            shell = environ.get("SHELL")
```

Comment:
`$SHELL` isn't necessarily the right shell to use, see pypa/pipenv#1762 for more discussion. On the other hand, finding the actual shell in use is fraught and apparently doesn't work properly on Windows, so this may be the only sane thing to do.